### PR TITLE
Initialize and store PRv2 environment in cache

### PR DESCRIPTION
**Pull request by pgarg66**
#### Problem
Program runtime V2 environment is created every time a program is loaded. This can be optimized.

#### Summary of Changes
Store the environment in the cache.
Another refactor is needed to reuse the cached environment.

Fixes #

### CODE REVIEW

File: programs/bpf_loader/src/syscalls/mod.rs [hunk 1]
```diff
@@ -142,7 +142,7 @@ macro_rules! register_feature_gated_function {
     };
 }
 
-pub fn create_program_runtime_environment<'a>(
+pub fn create_program_runtime_v1_environment<'a>(
```
Comment (Lichtso) [inline programs/bpf_loader/src/syscalls/mod.rs:145]: I would put the `_v1` at the end.

File: programs/loader-v4/src/lib.rs [hunk 2]
```diff
@@ -104,15 +97,31 @@ pub fn load_program_from_account(
...
+    };
     let state = get_state(program.get_data())?;
     let programdata = program
         .get_data()
         .get(LoaderV4State::program_data_offset()..)
         .ok_or(InstructionError::AccountDataTooSmall)?;
     let loaded_program = LoadedProgram::new(
         &loader_v4::id(),
-        Arc::new(loader),
+        Arc::new(create_program_runtime_v2_environment(
```
Comment (Lichtso) [inline programs/loader-v4/src/lib.rs:121]: This entire function needs to be moved inside `Bank::load_program()` and then use `loaded_programs_cache.program_runtime_environment_v2`. But that can be done in a different PR.

File: runtime/src/bank.rs [hunk 3]
```diff
@@ -8010,6 +8010,17 @@ impl Bank {
                 loaded_programs_cache.program_runtime_environment_v1 =
                     Arc::new(program_runtime_environment_v1);
             }
+            let program_runtime_environment_v2 =
+                solana_loader_v4_program::create_program_runtime_v2_environment(
+                    &self.runtime_config.compute_budget.unwrap_or_default(),
+                    false, /* deployment */
```
Comment (Lichtso) [inline runtime/src/bank.rs:8016]: This is the parameter `debugging_features` not `deployment`, right?

Comment (pgarg66) [reply runtime/src/bank.rs:8016]: Yes, sorry, copypasta error on the comment.

File: programs/loader-v4/src/lib.rs (hunk 2 above)
Comment (pgarg66) [reply programs/loader-v4/src/lib.rs:121]: Yes, that's in my other PR, where it's integrating it to `bank::load_program`.

**Review by Lichtso — APPROVED** [review]

**Pull request merged**